sdk/objc/components/audio/RTCAudioSession.mm

```diff
@@ -494,9 +496,16 @@ - (void)handleRouteChangeNotification:(NSNotification *)notification {
     case AVAudioSessionRouteChangeReasonCategoryChange:
       RTCLog(@"Audio route changed: CategoryChange to :%@", self.session.category);
       {
-        BOOL newValue = [self sessionCategoryIsRecordingEnabled];
-        if (_isRecordingEnabled != newValue) {
-          _isRecordingEnabled = newValue;
+        // webRTCConfiguration is used by configureWebRTCSession
+        RTC_OBJC_TYPE(RTCAudioSessionConfiguration) *webRTCConfig = [RTC_OBJC_TYPE(RTCAudioSessionConfiguration) webRTCConfiguration];
+        webRTCConfig.category = _session.category;
+        webRTCConfig.categoryOptions = _session.categoryOptions;
+        webRTCConfig.mode = _session.mode;
+
+        if (![_session.category isEqualToString:_activeCategory]) {
```

Comment:
should we get rid of `_isRecordingEnabled` then?

Reply:
Maybe we don't need it...